app/frontend/components/pages/accounts/accounts.tsx

```diff
@@ -0,0 +1,286 @@
+import {Fragment, h} from 'preact'
+import {connect} from '../../../helpers/connect'
+import actions from '../../../actions'
+import range from '../../../../frontend/wallet/helpers/range'
+import printAda from '../../../helpers/printAda'
+import {Lovelace} from '../../../state'
+import {AdaIcon} from '../../common/svg'
+import Alert from '../../common/alert'
+import SendTransactionModal from './sendTransactionModal'
+import DelegationModal from './delegationModal'
+import tooltip from '../../common/tooltip'
+
+const Account = ({
+  i,
+  account,
+  firstAddressPerAccount,
```

Comment:
I find this a bit cumbersome TBH, I would just pass the full list of accounts, the tile's account index, selected account index and do the address retrieval/transformations needed inside, removing the concern of constructing a mapping of accounts to addresses by the outside logic

Reply:
I agree, I find it cumbersome as well, I forgot to remove this `address` argument and move it to actions.ts. Fixed.